RFCs/FS-1027-fsi-references.md

```diff
@@ -164,6 +162,18 @@ This will generate an error because `System.Whatever` version `4.1.0` will alrea
 
 We may consider using `AssemblyLoadContext` in some clever way in the future, but it's very difficult to get right and well outside the scope of what we need to support for a .NET Core 2.0 timeframe.
 
+## Handler resolution
+
+FSI/Design time support will look at the following places in order:
+
+* if the script is a physical file, check current folder and browse all parent folders looking for .fsharp/fsi-extensions folder in each one
+* look into ~/.fsharp/fsi-extensions
+* look into .fsharp/fsi-extensions folder next to fsi.exe
+
+gather all the distinct dll names, order of precedence favorising those in the same order shown above, and load them in the process if their assembly contains an arbitrary attribute (resolved by name rather than dependency on external library) and types marked with same attributes.
```

Comment:
@dsyme 

for (1)

would it be ok to just look at the dlls in that folder? for paket we don't have (nor want) assembly reference to paket, the dll can load the assembly and/or work through reflection also.

it seems those entry points should be as lightweight as possible, and yes they can rely on having a subfolder in that same location, but does it need to be strong reference? isn't it better to not have that extension folder become a jungle of dll/folders?

We can also reintroduce a naming pattern, in current PR we have this: `"*DependencyManager*.dll"`, to not have to scan every single file, please advise and yes we can also browse subfolders (one level only?) if you like.

for (2) yes will do.

for (3):

>  We should specify whether this means the DLLs are loaded multiple times from different locations as you open and work with different scripts.

good point, that should be discussed, but the behaviour has somehow to match what we will have in FSI, this is one reason to browse folders up (like paket users do with .paket at root of repository), but maybe for tooling we should actually mandate a single or more restricted location instead?

for (4) ok.